iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/OpenHouseInternalCatalog.java

```diff
@@ -218,10 +221,28 @@ public void purgeSoftDeletedTables(String databaseId, String tableId, long purge
         databaseId,
         tableId,
         purgeAfterMs);
-
     houseTableRepository.purgeSoftDeletedTables(databaseId, tableId, purgeAfterMs);
   }
 
+  public void restoreTable(String databaseId, String tableId, long deletedAtMs) {
+    log.info(
+        "Restoring soft deleted table for databaseId: {}, tableId: {}, deletedAtMs: {}",
+        databaseId,
+        tableId,
+        deletedAtMs);
+    try {
+      houseTableRepository.restoreTable(databaseId, tableId, deletedAtMs);
+    } catch (HouseTableNotFoundException e) {
+      throw new NoSuchUserTableException(
```

Comment:
Is this the right exception here? Could this also happen if table was already restored?